File: Sources/Api/ConfigurationAPI.m
//
//  ConfigurationAPI.m
//  LoginRadiusSDK
//
//  Created by LoginRadius on 14/12/17.
//

#import "ConfigurationAPI.h"

@implementation ConfigurationAPI


+ (instancetype)configInstance{
    static dispatch_once_t onceToken;
    static ConfigurationAPI *instance;
    
    dispatch_once(&onceToken, ^{
        instance = [[ConfigurationAPI alloc] init];
    });
    
    return instance;
}


- (void)getConfigurationSchema:(LRAPIResponseHandler)completion {
    
    [[LoginRadiusREST configInstance] sendGET:@"ciam/appInfo"
                               queryParams:@{
                                             @"apikey": [LoginRadiusSDK apiKey]
                                             }
                         completionHandler:completion];
    
}
@end

File: Sources/ConfigurationAPI.swift
//
//  ConfigurationAPI.swift
//  LoginRadiusSwift SDK
//
//  Created by Megha Agarwal.
//  Copyright © 2023 LoginRadius Inc. All rights reserved.
//

import Foundation

public class ConfigurationAPI {
    
    public static let shared = ConfigurationAPI()
    public static let configInstance = ConfigurationAPI.shared
    public init() {}
    
    public func getConfigurationSchema(completion: @escaping (LRAPIResponseHandler)) {
        
        
        LoginRadiusREST.configInstance.sendGET(url: "ciam/appInfo", queryParams: ["apikey": LoginRadiusSDK.sharedInstance.apiKey], completionHandler: completion)
        
    }
}
